kafka/producer/kafka.py

```diff
@@ -450,6 +450,9 @@ def __del__(self):
     def close(self, timeout=None):
         """Close this producer.
 
+        If a non-zero timeout is passed, this will also attempt to flush the buffer before
```

Comment:
I think it would be better to say:

Blocks until all internally buffered records are sent and any required acks are received, or until a specified timeout is reached.

Reply:
No problems with changing the wording, although the code reads to my (unfamiliar) eye like a zero timeout would instantly skip the entire "graceful shutdown" block. A `0` and a `None` timeout are different: None fetches a default timeout (which could in turn be 0), and `0` just skips the entire graceful shutdown block and immediately closes. That seems slightly at odds with the proposed documentation?

Notably, my suggested documentation does not make the difference between 0 and None clear, either. This is more a bugreport masking as a PR, than a really well thought out proposal :) very open to further suggestions like yours.